src/it/scala/net/snowflake/spark/snowflake/benchmarks/PerformanceSuite.scala

```diff
@@ -0,0 +1,174 @@
+  protected final var outputFormat: String = ""
+
+  protected final var runTests: Boolean = true
+
+  // Maintain session state to make sure it is restored upon finishing of suite
+  protected final var sessionStatus: Boolean = false
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+
+    sessionStatus = getSessionStatus()
+
+    try {
+      runOption = getConfigValue("runOption")
+      outputFormat = getConfigValue("outputFormat")
+
+      for ((k, v) <- requiredParams)
+        requiredParams.put(k, getConfigValue(k).toLowerCase)
+
+    } catch {
+      case t: TestFailedException => {
+        if (t.getMessage contains "Config file needs to contain") {
+          runTests = false
+          val reqParams = requiredParams.keySet.mkString(", ")
+          println(
+            s"""One or more required parameters for running the benchmark suite was missing: " +
+              "runOption, outputFormat, $reqParams. Skipping ${getClass.getSimpleName}.""")
+        } else throw t
+
+      }
+      case e: Exception => throw e
+    }
+
+    if (runTests)
+      verifyParams()
+  }
+
+  protected final def verifyParams(): Unit = {
+
+    val argCheckMap = mutable.LinkedHashMap() ++= acceptedArguments
+    argCheckMap.put("runOption", runOptionAccepted)
+    argCheckMap.put("outputFormat", outputFormatAccepted)
+
+    val fullParams = mutable.LinkedHashMap() ++= requiredParams
+    fullParams.put("runOption", runOption)
+    fullParams.put("outputFormat", outputFormat)
+
+    for ((param, acceptedSet) <- argCheckMap) {
+      val argValue = fullParams
+        .getOrElse(param, fail(s"Required parameter $param missing."))
+      if (!acceptedSet.contains(argValue))
+        fail(
+          s"""Value $argValue not accepted for parameter $param. Accepted values are: ${acceptedSet
+            .mkString(", ")} """)
+    }
+  }
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+  }
+
+  override def afterAll(): Unit = {
+    SnowflakeConnectorUtils.setPushdownSession(sparkSession, sessionStatus)
+  }
+
+  /**
+    * Run the query using the given runtime configurations (S3-direct, with pushdown, without pushdown, etc.,
+    * outputting using also the provided config value.
+    */
+  protected final def testQuery(query: String,
+                                name: String = "unnamed"): Unit = {
+
+    val failedMessage = s"""Failed: $name. No result generated."""
+
+    // Skip if invalid configs
+    if (!runTests) return
+
+    var result: String = ""
+
+    if (runOption == "both") {
+      result = runWithSnowflake(query, name, true).getOrElse(failedMessage)
+      outputResult(s"""${name.toUpperCase}, with full pushdowns: """ + result)
+
+      result = runWithSnowflake(query, name, false).getOrElse(failedMessage)
+      outputResult(
+        s"""${name.toUpperCase}, with only filter/proj pushdowns: """ + result)
+    } else if (runOption == "full-only") {
```

Comment:
I'd remove the branch above, and simply make this one 
  if  (runOption == "full-only" or runOption == "both")
and similar for the else below.